resources/server/sv_sellout.ts

```diff
@@ -35,6 +38,35 @@ async function addListing(
   ]);
 }
 
+async function deleteListing(listingId: number, identifier: string): Promise<void> {
+  const query = 'DELETE FROM npwd_sellout_listing WHERE id = ? AND identifier = ?';
+
+  await pool.query(query, [listingId, identifier]);
+}
+
+async function getListing(listingId: number): Promise<MarketplaceListing> {
+  const query = `SELECT * FROM npwd_sellout_listing WHERE id = ?`;
+  const [results] = await pool.query(query, [listingId]);
+  const listings = <MarketplaceListing[]>results;
+  const listing = listings[0];
+
+  return listing;
+}
+
+async function reportListing(listingId: number, profile: string): Promise<void> {
+  const query = `INSERT INTO npwd_marketplace_reports (listing_id, profile) VALUES (?, ?)`;
+
+  await pool.query(query, [listingId, profile]);
+}
+
+async function doesReportExist(listingId: number, profile: string): Promise<boolean> {
+  const query = `SELECT * FROM npwd_marketplace_reports WHERE listing_id = ? AND profile = ?`;
```

Comment:
You can `SELECT COUNT(*)` and just parse that result, it may be more performant. I'm not sure why I didn't do that on the twitter one.

Probably not super necessary since we're already filtering by id.